src/workerd/api/web-socket.c++

```diff
@@ -633,37 +616,44 @@ size_t countBytesFromMessage(const kj::WebSocket::Message& message) {
 }
 }
 
-kj::Promise<void> WebSocket::pumpAfterFrontOutputLock(
-    IoContext& context, OutgoingMessagesMap& outgoingMessages, kj::WebSocket& ws) {
-  GatedMessage gatedMessage =
-      outgoingMessages.release(*outgoingMessages.ordered().begin());
-  auto size = countBytesFromMessage(gatedMessage.message);
+kj::Promise<void> WebSocket::pump(
+    IoContext& context, OutgoingMessagesMap& outgoingMessages, kj::WebSocket& ws, Native& native) {
+  native.isPumping = true;
```

Comment:
Can you `KJ_REQUIRE(!native.isPumping)` before setting it true here, just to make sure there's no way multiple pumps could be running in parallel?